common/persistence/sql/sqlShardPersistence.go

```diff
@@ -0,0 +1,359 @@
+shards 
+SET
+shard_id = :shard_id,
+owner = :owner,
+range_id = :range_id,
+stolen_since_renew = :stolen_since_renew,
+updated_at = :updated_at,
+replication_ack_level = :replication_ack_level,
+transfer_ack_level = :transfer_ack_level,
+timer_ack_level = :timer_ack_level,
+cluster_transfer_ack_level = :cluster_transfer_ack_level,
+cluster_timer_ack_level = :cluster_timer_ack_level,
+domain_notification_version = :domain_notification_version
+WHERE
+shard_id = :shard_id AND
+range_id = :old_range_id
+`
+
+	lockShardSQLQuery   = `SELECT 1 FROM shards WHERE shard_id = ? FOR UPDATE`
+	updateLeaseSQLQuery = `SELECT 1 FROM shards WHERE shard_id = ? AND range_id = ?`
+)
+
+func NewShardPersistence(username, password, host, port, dbName string) (persistence.ShardManager, error) {
+	var db, err = sqlx.Connect("mysql",
+		fmt.Sprintf(Dsn, username, password, host, port, dbName))
+	if err != nil {
+		return nil, err
+	}
+
+	return &sqlShardManager{
+		db: db,
+	}, nil
+}
+
+func (m *sqlShardManager) Close() {
+	if m.db != nil {
+		m.db.Close()
+	}
+}
+
+func (m *sqlShardManager) CreateShard(request *persistence.CreateShardRequest) error {
+	_, err := m.GetShard(&persistence.GetShardRequest{request.ShardInfo.ShardID})
+	if err == nil {
+		// The shard already exists.
+		return &persistence.ShardAlreadyExistError{
+			Msg: fmt.Sprintf("CreateShard operation failed. Shard already exists. ShardId: %v, RangeId: %v", request.ShardInfo.ShardID, request.ShardInfo.RangeID),
+		}
+	} else {
+		switch err.(type) {
+		case *workflow.EntityNotExistsError:
+			row, err := shardInfoToShardsRow(*request.ShardInfo)
+			if err != nil {
+				return &workflow.InternalServiceError{
+					Message: fmt.Sprintf("CreateShard operation failed. Error: %v", err),
+				}
+			}
+
+			if _, err := m.db.NamedExec(createShardSQLQuery, row); err != nil {
+				return nil
+			}
+		default:
+			return &workflow.InternalServiceError{
+				Message: fmt.Sprintf("CreateShard operation failed. Failed to verify whether a shard of ID %v already existed. Error: %v", request.ShardInfo.ShardID, err),
+			}
+		}
+	}
+
+	return nil
+}
+
+func (m *sqlShardManager) GetShard(request *persistence.GetShardRequest) (*persistence.GetShardResponse, error) {
+	stmt, err := m.db.PrepareNamed(getShardSQLQuery)
+	if err != nil {
+		return nil, &workflow.InternalServiceError{
+			Message: fmt.Sprintf("GetShard operation failed. Failed to prepare statement. ShardId: %v. Error: %v", request.ShardID),
+		}
+	}
+
+	var row shardsRow
+	err = stmt.Get(&row, struct {
+		ShardID int `db:"shard_id"`
+	}{request.ShardID})
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return nil, &workflow.EntityNotExistsError{
+				Message: fmt.Sprintf("GetShard operation failed. Shard with ID %v not found. Error: %v", request.ShardID, err),
+			}
+		}
+		return nil, &workflow.InternalServiceError{
+			Message: fmt.Sprintf("GetShard operation failed. Failed to get record. ShardId: %v. Error: %v", request.ShardID, err),
+		}
+	}
+
```

Comment:
The following would be simpler to write for queries:
```
	var row shardsRow
	err := m.db.Get(row, getShardSQLQuery, request.ShardID)
	if err != nil {
		if err == sql.ErrNoRows {
			return nil, &workflow.EntityNotExistsError{
				Message: fmt.Sprintf("GetShard operation failed. Shard with ID %v not found. Error: %v", request.ShardID, err),
			}
		}
		return nil, &workflow.InternalServiceError{
			Message: fmt.Sprintf("GetShard operation failed. Failed to get record. ShardId: %v. Error: %v", request.ShardID, err),
		}
	}
```

Then your sql query should look like
```
SELECT shard_id,
       owner,
       range_id,
       stolen_since_renew,
       updated_at,
       replication_ack_level,
       transfer_ack_level,
       timer_ack_level,
       cluster_transfer_ack_level,
       cluster_timer_ack_level,
       domain_notification_version
FROM   shards
WHERE  shard_id = ?
```